oada-core/http-handler/oada-srvc-http-handler/websockets.ts

```diff
@@ -0,0 +1,434 @@
+                                rev,
+                                request.headers['x-oada-rev']
+                            )
+                            // Next, feed changes to client
+                            const newChanges = await changes.getChangesSinceRev(
+                                resourceId,
+                                request.headers['x-oada-rev']
+                            )
+                            newChanges.forEach(change =>
+                                sendChange({
+                                    requestId: [msg.requestId],
+                                    resourceId,
+                                    path_leftover,
+                                    change
+                                })
+                            )
+                        }
+                    } else {
+                        sendResponse({
+                            requestId: msg.requestId,
+                            status: 'success'
+                        })
+                    }
+                    break
+
+                case 'delete':
+                    if (parts.length === 3) {
+                        // it is a resource
+                        emitter.removeAllListeners(resourceId)
+                    }
+                default:
+                    sendResponse({
+                        requestId: msg.requestId,
+                        status: res.status,
+                        headers: res.headers,
+                        data: res.data
+                    })
+            }
+        }
+    })
+
+    const interval = setInterval(function ping () {
+        wss.clients.forEach(function each (sock) {
+            const socket = sock as Socket // TODO: Better way to do this?
+            if (socket.isAlive === false) {
+                return socket.terminate()
+            }
+
+            socket.isAlive = false
+            socket.ping()
+        })
+    }, 30000)
+
+    wss.on('close', function close () {
+        clearInterval(interval)
+    })
+}
+
+const writeResponder = new Responder({
+    consumeTopic: config.get('kafka:topics:httpResponse') as string,
+    group: 'websockets'
+})
+
+type WriteResponse = {
+    msgtype: 'write-response'
+    code: 'success'
+    resource_id: string
+    path_leftover: string
+    _rev: number
+}
+function checkReq (req: KafkaRequest): req is WriteResponse {
+    return req.msgtype === 'write-response' && req.code === 'success'
+}
+// Listen for successful write requests to resources of interest, then emit an event
+writeResponder.on('request', async function handleReq (req) {
+    if (!checkReq(req)) {
+        return
+    }
+
+    trace('@@@@@@@@@@@@@@@', req.resource_id)
+    try {
+        const change = await changes.getChange(req.resource_id, req._rev)
+        trace('Emitted change for:', req.resource_id, change)
+        emitter.emit(req.resource_id, {
+            path_leftover: req.path_leftover,
+            change
+        })
```

Comment:
We might want to add a code that excludes changes that occurred above path_leftover. Maybe I will add the code after this pull request is merged in.